v1/plugins/status/plugin.go

```diff
@@ -418,15 +449,39 @@ func (p *Plugin) loop(ctx context.Context) {
 }
 
 // readBundleStatus is a non-blocking read to make sure the latest status is received
-func (p *Plugin) readBundleStatus() {
-	select {
-	case status := <-p.bulkBundleCh:
-		p.lastBundleStatuses = status
-	case status := <-p.bundleCh:
+func (p *Plugin) readBundleStatus() bool {
+	var changed bool
+	if len(p.pluginStatusCh) != 0 {
+		p.lastPluginStatuses = <-p.pluginStatusCh
```

Comment:
Depending on the status of `loop()` could this - and below - turn into a blocking read if `loop()` manages to squeeze in a read between `if len(p.pluginStatusCh) != 0` and here? Or are we confident this won't happen?

Should these be individual `select` blocks? E.g.

```go
select {
case status <- p.pluginStatusCh:
   p.lastPluginStatuses = status
   changed = true
default:
}

select {
case status <- p.bulkBundleCh:
   p.lastBundleStatuses = status
   changed = true
default:
}

etc ...
```

Reply:
Good idea! Updated to use `select` instead to avoid any possibility of blocking.